### ISSUE TITLE
[Repository] refactor helm to use managed-by annotation instead of system flag

### ISSUE THREAD

**Issue by dlaloue-vmware** [vmware-tanzu/kubeapps#4870]
For repositories, secrets used for authentication can be either user managed or kubeapps managed.
The current mechanism is based on a system flag to enable one more or the other mode. This limits the flexibility to have some users use one way (e.g. cli) or the other (e.g. UI) in a single Kubeapps installation.

this issue is to refactor the helm plugin.

**Comment by gfichtenholt** [comment]
I don't think this should be closed. The current implementation does not use "managed-by" annotation as the title of this issue suggests, but rather relies on a hack
https://github.com/vmware-tanzu/kubeapps/blob/e5aa78415f8c8620324e5ebd379eb0294163a219/cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/repositories_auth.go#L669

Implement the annotation, then close the issue, IMHO
